### PR TITLE
Feature/add players implementation

### LINKED ISSUE #10: REQ-009: Player Core System and Score Management
# Player Core System and Score Management Implementation

## Description
Create a doubly linked list system for players, including score management based on game difficulty and handling of duplicate nicknames.

### Data Structure
```c
typedef struct Player {
    int id;
    char nickname[MAX_STRING];
    float maxScore;
    struct Player* next;
    struct Player* prev;
} Player;
```

### Scoring System
* Points per correct answer:
  - Easy Mode: 0.5 points
  - Normal Mode: 1.0 points
  - God Mode: 2.0 points

### Required Functions
```c
Player* createPlayer(int id, char *nickname, float score);
void insertSortedPlayer(Player **head, Player *newPlayer);
Player* findPlayerByNickname(Player *head, char *nickname);
void updatePlayerIfHigherScore(Player **head, char *nickname, float newScore);
```

### Core Features
* Player Creation:
  - After difficulty selection
  - Unique ID assignment
  - Nickname input
* Score Management:
  - Calculate based on difficulty
  - Update highest scores
* Duplicate Handling:
  - Check existing nicknames
  - Keep highest score only
  - Update list order if needed
* Sorted List:
  - Maintain highest to lowest score order
  - Update order after score changes

## Acceptance Criteria
- [x] Player structure properly implemented
- [x] Creation function works correctly:
  - Proper ID assignment
  - Memory allocation
  - Initial score setup
- [x] Score calculation works:
  - Easy mode: 0.5 per question
  - Normal mode: 1.0 per question
  - God mode: 2.0 per question
- [x] Duplicate nickname handling:
  - Finds existing players
  - Updates if new score is higher
  - Maintains only highest score
- [x] Sorted insertion works:
  - Correct order by score
  - Proper link management
  - Handles all cases:
    * Empty list
    * Start/middle/end insertion
- [x] Memory management is proper
- [x] Double links maintained correctly

## Technical Notes
* Implement proper memory allocation/deallocation
* Handle all list edge cases
* Validate all inputs
* Maintain doubly linked integrity
* Document sorting algorithm
* Error handling implementation

## Integration Points
* Works with game difficulty selection
* Integrates with scoring system
* Maintains data consistency
* Prepares for persistence

### PR DESCRIPTION
**Pull request by J-P-T-0**
Player doubly licked list added to project with it's corresponding structure. Here all new players are stored once they've obtained a previous score. Several functions added in order to maintain the players with the highest scores at the beginning of the list and have them ordered downwards. The players are rearranged each time a player gets a new high score. Score system implementation for scoring multipliers. Nickname creation system to guarantee each new player is unique as well as their ID.

### CODE REVIEW

File: marathon_quiz.cpp [hunk 1]
```diff
@@ -191,3 +191,117 @@ void freeQuestions(struct Question* head) {
...
+    newPlayer->next=NULL;
+    newPlayer->prev=NULL;
+    return newPlayer;
+}
+
+//Insert new player in order from the highest score to the lowest
+void insertSortedPlayer(Player **head, Player *newPlayer, bool &inserted){
+    //Check for repeated nicknames
+    Player *current=*head;
+    if(current!=NULL){
```
Comment (gfdfranco) [inline marathon_quiz.cpp]: Please refacto this code. I think a better approach is create another function to return is a player is already in the list.

**Review by gfdfranco — CHANGES_REQUESTED** [review]
Please review the comments @J-P-T-0

File: marathon_quiz.cpp (hunk 1 above)
Comment (Nohelany) [reply marathon_quiz.cpp]: I agree with this

**Comment by Nohelany** [comment]
- Create another function to return if a player is already in the list.
- Create another function to free the players.
- Create a function to create a user, and call it in the play game function.

File: marathon_quiz.cpp (hunk 1 above)
Comment (J-P-T-0) [reply marathon_quiz.cpp]: Changes have been made

**Comment by J-P-T-0** [comment]
I accidentally wrote two commits in past tense :( when I noticed it was way too late and I couldn't change the names

File: marathon_quiz.cpp (hunk 1 above)
Comment (gfdfranco) [reply marathon_quiz.cpp]: Awesome !!

**Review by gfdfranco — APPROVED** [review]
LGTM

**Review by Nohelany — APPROVED** [review]

**Comment by Mr-Mauxerr** [comment]
LGTM

**Pull request merged**